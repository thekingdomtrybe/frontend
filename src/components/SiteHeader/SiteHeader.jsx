import React from 'react';
import PropTypes from 'prop-types';
import SiteLogo from '@/components/SiteLogo/SiteLogo';
import DropdownMenu from '../DropdownMenu/DropdownMenu';
import Navbar from '../Navbar/Navbar';
import Styles from './SiteHeader.module.scss';

function SiteHeader({
  live,
  variant,
}) {
  let className = Styles.header;

  switch (variant) {
    case 'blue': {
      className = `${className} ${Styles.blue}`;
      break;
    }
    case 'dark': {
      className = `${className} ${Styles.dark}`;
      break;
    }
    default:
      break;
  }

  const navLinks = [
    { label: 'ATTEND ONLINE', url: '/live' },
    { label: 'COMMUNITY', url: '/join-our-community' },
    { label: 'COUNSELLING', url: '/request-counselling' },
    { label: 'GIVE', url: '/give' },
  ];

  return (
    <header className={className}>
      <SiteLogo variant={variant} live={live} />
      <div className={Styles.nav}>
        <Navbar links={navLinks} variant={variant} />
      </div>
      <DropdownMenu
        variant={variant}
        menuItems={[
          { label: 'Join Our Community', url: '/join-our-community', tablet: true },
          { label: 'Book A Counselling Session', url: '/request-counselling', tablet: true },
          { label: 'Ask For Prayer', url: '/request-for-prayer' },
          { label: 'Share Your Testimony', url: '/share-a-testimony' },
          { label: 'Watch Past Sermons', url: '/browse-past-sermons' },
          { label: 'See Upcoming Events', url: '/upcoming-events' },
          { label: 'Join a Volunteer Group', url: '/volunteer' },
          { label: 'Give', url: '/give', tablet: true },
          { label: 'Login', url: '/login' },
          { label: 'Settings', url: '/settings', auth: true },
        ]}
      />
    </header>
  );
}

SiteHeader.propTypes = {
  variant: PropTypes.string,
  live: PropTypes.bool,
};

SiteHeader.defaultProps = {
  variant: 'light',
  live: false,
};

export default SiteHeader;
