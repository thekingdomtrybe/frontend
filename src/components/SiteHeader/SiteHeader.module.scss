.header {
  width: 100%;
  padding: 0 var(--spacing-3-half);
  height: var(--header-height);
  display: grid;
  justify-content: space-between;
  align-items: center;
  grid-template-columns: max-content max-content;
  transition: var(--timing-medium) all;
  position: sticky;
  top: 0;
  background-color: var(--white);
  z-index: 100;
}

.menus {
  display: grid;
  grid-template-columns: max-content max-content;
}

.dark {
  background-color: var(--dark);
}

.blue {
  background-color: var(--blue);
}

.live {
  background-color: var(--live-gray);
}

.nav {
  display: none;
}

.light-border {
  border-bottom: calc(var(--size-1-quarter)) solid var(--alpha-gray-3);
}

@media (width >= 768px) {
  .header {
    padding: 0 var(--spacing-2);
  }

  .light-border {
    border-bottom: calc(var(--size-1-quarter)) solid var(--alpha-silver-9);
  }
}

@media (width >= 850px) {
  .nav {
    display: block;
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
  }

  .live {
    .nav {
      display: none;
    }
  }
}
