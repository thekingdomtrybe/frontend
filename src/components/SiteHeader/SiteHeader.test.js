import renderer from 'react-test-renderer';
import SiteHeader from './SiteHeader.jsx';

it('renders correctly', () => {
  const component = renderer.create(<SiteHeader />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree = null;
  // manually trigger the callback
  // renderer.act(() => {
  //   tree.props.onMouseLeave();
  // });

  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});
