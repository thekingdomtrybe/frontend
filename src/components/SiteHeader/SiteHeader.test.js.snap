// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="logo"
  >
    <img
      alt=""
    />
    <span>
      THE KINGDOM TRYBE
    </span>
  </div>
</header>
`;
